internal/entitlement/postgresadapter/entitlement.go

```diff
@@ -145,8 +145,15 @@ func (a *entitlementDBAdapter) GetEntitlementsOfSubject(ctx context.Context, nam
 }
 
 func (a *entitlementDBAdapter) ListEntitlements(ctx context.Context, params entitlement.ListEntitlementsParams) ([]entitlement.Entitlement, error) {
-	query := withLatestUsageReset(a.db.Entitlement.Query().
-		Where(db_entitlement.Namespace(params.Namespace)))
+	query := a.db.Entitlement.Query()
+
+	if len(params.Namespaces) > 1 {
+		query.Where(db_entitlement.NamespaceIn(params.Namespaces...))
+	} else if len(params.Namespaces) == 1 {
+		query.Where(db_entitlement.NamespaceEQ(params.Namespaces[0]))
+	}
```

Comment:
nit: I think it's enough to just use the In

```suggestion
	if len(params.Namespaces) > 0 {
		query.Where(db_entitlement.NamespaceIn(params.Namespaces...))
	}
```